resources/modes/print/print.js

```diff
@@ -30,10 +30,21 @@ $(function() {
           .text(robopaint.t('common.action.start'));
         $('#buttons button.normal').prop('disabled', false); // Enable options
         $('#cancel').prop('disabled', true); // Disable the cancel print button
+        $("#processing-dialog").hide();
         break;
     }
   });
 
+  if (robopaint.cncserver.api.server.domain == 'localhost') {
```

Comment:
This entire `if` block seems to be stolen (and stealing) parts that are [already being setup](https://github.com/evil-mad/robopaint/blob/master/resources/scripts/robopaint.mode.svg.js#L63). What you likely need to do is actually have this code live in [`robopaint.mode.svg.js`](/evil-mad/robopaint/blob/master/resources/scripts/robopaint.mode.svg.js), and do its work within the buffer update event function already there. This functionality really makes the most sense as something that will work for any mode doing work with SVGs, and therefore makes sense to live there in this SVG "library" we're providing there.